src/main/java/com/google/devtools/build/lib/sandbox/SandboxStash.java

```diff
@@ -143,8 +143,12 @@ private void stashSandboxInternal(
     try {
       stashPath.createDirectory();
       Path stashPathExecroot = stashPath.getChild("execroot");
+      if (isTestAction(mnemonic)) {
+        treeDeleter.deleteTree(path.getRelative("execroot/" + environment.get("TEST_WORKSPACE") + "/_tmp"));
+      }
       path.getChild("execroot").renameTo(stashPathExecroot);
       if (isTestAction(mnemonic)) {
+        // We only do this after the rename operation has succeeded
```

Comment:
Not related to a change made in this PR, but the `synchronized` block in line 136 looks unnecessary.

Reply:
Done.